### PR TITLE
Add e2e test for setup actions

### PR DESCRIPTION
**Pull request by fgimenez**
Towards https://github.com/giantswarm/giantswarm/issues/2070

Initial test for `setup` actions, mostly adding the required helpers for other tests.

### CODE REVIEW

File: integration/test/basic/basic_test.go [hunk 1]
```diff
@@ -3,8 +3,178 @@
...
+	serverAddr := strings.Split(hostURL.Host, ":")[0]
+	port := vaultSvc.Spec.Ports[0].NodePort
+	addr := fmt.Sprintf("http://%s:%d", serverAddr, port)
+
+	return addr, nil
+}
+
+func waitForVault(client *vaultclient.Client) error {
+	o := func() error {
+		req := client.NewRequest("HEAD", "/sys/health")
```
Comment (rossf7) [inline integration/test/basic/basic_test.go]: Nice I like the use of the health endpoint for this.

File: integration/test/basic/basic_test.go [hunk 2]
```diff
@@ -3,8 +3,178 @@
...
+
+	vaultSvc, err := f.K8sClient().
+		CoreV1().
+		Services("default").
+		Get("vault", meta_v1.GetOptions{})
+	if err != nil {
+		return "", microerror.Mask(err)
+	}
+
+	// we will access Vault service from the test container using the k8s API
```
Comment (rossf7) [inline integration/test/basic/basic_test.go]: Nit but can you start the comment with caps.

**Review by rossf7 — APPROVED** [review]
LGTM

File: integration/test/basic/basic_test.go (hunk 2 above)
Comment (fgimenez) [reply integration/test/basic/basic_test.go]: Sure, done.

File: integration/test/basic/project.yaml [hunk 3]
```diff
@@ -0,0 +1,5 @@
+version: 1
+
+test:
+  env:
+    - "VAULT_TOKEN=myToken"
```
Comment (kopiczko) [inline integration/test/basic/project.yaml]: Is it used?

Comment (fgimenez) [reply integration/test/basic/project.yaml]: Indeed, I've proposed https://github.com/giantswarm/e2e-harness/pull/92 to make things more clear.

Comment (kopiczko) [reply integration/test/basic/project.yaml]: Sorry I got confused. So it can be removed?

Comment (fgimenez) [reply integration/test/basic/project.yaml]: Proposed here https://github.com/giantswarm/certctl/pull/70.

Comment (kopiczko) [reply integration/test/basic/project.yaml]: Ack. I'd prefer if the project.yaml changes were introduced with #70. That would be easier to follow I think.
Thanks for explanations.

**Pull request merged**